From 55d47cff2f5944f3f70fbee5f2b41beccbed2762 Mon Sep 17 00:00:00 2001
From: Zhu Yi <yi.zhu@intel.com>
Date: Sat, 30 Jun 2007 11:10:59 +0800
Subject: [PATCH] mac80211: [HT] fix wrong param used for ieee80211_ht_agg_queue_remove

Fix wrong param used for ieee80211_ht_agg_queue_remove() and
compile warnings.

Signed-off-by: Zhu Yi <yi.zhu@intel.com>
---
 net/mac80211/ieee80211_sta.c |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/net/mac80211/ieee80211_sta.c b/net/mac80211/ieee80211_sta.c
index 72d1ce4..924c871 100644
--- a/net/mac80211/ieee80211_sta.c
+++ b/net/mac80211/ieee80211_sta.c
@@ -2779,7 +2779,7 @@ static void ieee80211_sta_process_addba_resp(struct net_device *dev,
 				printk (KERN_ERR "HW problem - can not stop aggergation "
       					"for tid %d\n",tid);
 			/* remove the queue for this aggregation */
-      			ieee80211_ht_agg_queue_remove(local,sta->addr,tid);
+      			ieee80211_ht_agg_queue_remove(local, sta, tid);
 		}
 		spin_lock_bh(&sta->ht_ba_mlme.agg_data_lock_tx);
 		sta->ht_ba_mlme.tid_agg_info_tx[tid].state = HT_AGG_STATE_IDLE;
@@ -2931,7 +2931,7 @@ static void ieee80211_sta_process_delba(struct net_device *dev, struct ieee80211
 				printk (KERN_ERR "HW problem - can not stop"
 									" aggergation for tid %d\n",tid);
 			/* remove the queue for this aggregation */
-			ieee80211_ht_agg_queue_remove(local,sta,tid);
+			ieee80211_ht_agg_queue_remove(local, sta, tid);
 		}
 
 		spin_lock_bh(&sta->ht_ba_mlme.agg_data_lock_tx);
@@ -2983,7 +2983,7 @@ static void ieee80211_sta_process_delba(struct net_device *dev, struct ieee80211
  		if(rc)
  			printk (KERN_ERR "HW problem - can not stop aggergation for tid %d\n",tid);
  		/* remove the queue for this aggregation */
-		ieee80211_ht_agg_queue_remove(local,sta,tid);
+		ieee80211_ht_agg_queue_remove(local, sta, tid);
  	}
  	/* send a delBA */
  	ieee80211_send_delba(sta->dev, sta->addr, tid, 1 /* WLAN_BACK_INITIATOR */, WLAN_REASON_QSTA_NOT_USE);
@@ -3006,7 +3006,8 @@ static void ieee80211_sta_process_delba(struct net_device *dev, struct ieee80211
  	* only one argument, and ieee80211_local is needed here */
  	int *ptid = (int *)data;
  	int *timer_to_id = ptid-*ptid;
- 	struct sta_info *temp_sta = container_of(timer_to_id,struct sta_info,timer_to_tid);
+ 	struct sta_info *temp_sta = container_of(timer_to_id, struct sta_info,
+						 timer_to_tid[0]);
  	struct ieee80211_local *local = temp_sta->local;
  	struct sta_info *sta;
  	u16 tid = (u16)*ptid;
-- 
1.5.2

